src/inmanta/module.py

```diff
@@ -1988,10 +1990,13 @@ def verify_python_requires(self) -> None:
         """
         Verifies no incompatibilities exist within the Python environment with respect to installed module v2 requirements.
         """
-        if not env.ActiveEnv.check(in_scope=re.compile(f"{ModuleV2.PKG_NAME_PREFIX}.*")):
-            raise CompilerException(
-                "Not all installed modules are compatible: requirements conflicts were found. Please resolve any conflicts"
-                " before attempting another compile. Run `pip check` to check for any incompatibilities."
+        constraints: List[Requirement] = [Requirement.parse(item) for item in self.collect_python_requirements()]
+        constraints_for_marker = filter(lambda c: not (c.marker and not c.marker.evaluate()), constraints)
+        if not env.ActiveEnv.check(constraints=constraints_for_marker):
+            raise env.ConflictingRequirements(
+                "Module dependency resolution conflict: a module dependency constraint "
+                "was violated by another module. This most likely indicates an incompatibility between "
+                "two or more of the installed modules. You can get more details on the issue with 'pip check'."
```

Comment:
The `pip check` suggestion will only work for V2 module dependencies. I see the `check()` method writes warning log lines with more information about the conflict. Wouldn't it be better to refer to those. Or even better:

* Make the `check()` method raise the `env.ConflictingRequirements` exception and add an argument to the exception that contains the details regarding the violations.
* Catch the `env.ConflictingRequirements` here and turn it into a `CompilerException` with the above-mentioned message that also includes the specific details of the conflicts that exist.

Wdyt?

Reply:
I like the suggestion. An `inmanta venv check` command or something could be nice when the user is trying to fix the issue. But depending on if/when we implement #4279 that might not be worth the effort.

Reply:
I can try to do that. The more info in the exception the better I guess? should I keep the logged warning or removed it?

Reply:
I don't think it's useful to write the same information to two different streams. I would remove the log line if we incorporate that information into the error message.